src/components/AddressSearch/types.ts

```diff
@@ -13,13 +13,13 @@ type CurrentLocationButtonProps = {
 };
 
 type RenamedInputKeysProps = {
-    street: string;
-    street2: string;
-    city: string;
-    state: string;
-    lat: string;
-    lng: string;
-    zipCode: string;
+    street?: string;
```

Comment:
Maybe you can try to type `inputKeys` that you pass to AdressForm => this way you should be able to simplify `inputKeys` type in the AddressForm and maybe even reuse `Address` type here